plugins/lttng-utils/copy.c

```diff
@@ -159,25 +149,21 @@ int get_stream_event_context_unsigned_int_field_value(FILE *err,
 	}
 
 	field_type = bt_ctf_field_get_type(field);
-	if (!field_type) {
-		fprintf(err, "[error] %s in %s:%d\n", __func__,
-				__FILE__, __LINE__);
-		goto error;
-	}
+	assert(field_type);
 
 	if (bt_ctf_field_type_get_type_id(field_type) != BT_CTF_FIELD_TYPE_ID_INTEGER) {
-		fprintf(err, "[error] %s in %s:%d\n", __func__,
-				__FILE__, __LINE__);
+		BT_LOGE("Wrong type, expected integer for field=%s", field_name);
 		goto error;
 	}
 
 	if (bt_ctf_field_type_integer_get_signed(field_type) != 0) {
-		fprintf(err, "[error] %s in %s:%d\n", __func__,
-				__FILE__, __LINE__);
+		BT_LOGE("Wrong type, expected unsigned integer for field=%s",
+				field_name);
 		goto error;
 	}
 
 	ret = bt_ctf_field_unsigned_integer_get_value(field, value);
+	assert(!ret);
```

Comment:
Same as before, should we take care on unset fields in the plugins are can we expect the input to always have all the fields set ?